### PR TITLE
add nfs storageClass example

### PR DESCRIPTION
**Pull request by chenxuc**
fixed: #28924

### CODE REVIEW

**Comment by tengqm** [comment]
I'd suggest we add a little bit description for the parameters, which are actually the fields defined for `NFSVolumeSource`, see here: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#nfsvolumesource-v1-core

Pointing the docs to examples is acceptable, but it may also become misleading because such an example may get misinterpreted as the "official" one.

**Comment by chenxuc** [comment]
> I'd suggest we add a little bit description for the parameters, which are actually the fields defined for `NFSVolumeSource`, see here: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#nfsvolumesource-v1-core
> 
> Pointing the docs to examples is acceptable, but it may also become misleading because such an example may get misinterpreted as the "official" one.

Thanks for the suggestions. 
I've added a NFS section to explicitly point out they are just examples instead of official ones.

The parameters in `NFSVolumeSource` seems just for PV/PVC instead of StorageClass. The NFS StorageClass's parameters varies depending on the external provisioner. So it'd better we don't describe specific NFS StorageClass parameters in this page.

**Comment by tengqm** [comment]
`parameters` is a channel for the StorageClass to pass additional settings to the PV provisioner. For different types of PVs, the parameters differ. As for NFS volumes, you will refer to the`NFSVolumeSource` for the parameters that can be passed from the StorageClass to the provisioner. For other volume types, the same applies. There is no dedicated data structures for defining the interface.

**Comment by tengqm** [comment]
`parameters` is a channel for the StorageClass to pass additional settings to the PV provisioner. For different types of PVs, the parameters differ. As for NFS volumes, you will refer to the`NFSVolumeSource` for the parameters that can be passed from the StorageClass to the provisioner. For other volume types, the same applies. There is no dedicated data structures for defining the interface.

**Comment by chenxuc** [comment]
Thanks @tengqm for the explaination. I've updated the doc with an example for nfs StorageClass and some description for the optional parameters.

File: content/en/docs/concepts/storage/storage-classes.md [hunk 1]
```diff
@@ -423,6 +423,29 @@ parameters:
...
+  server: nfs-server.example.com
+  path: /share
+  readOnly: false
+```
+
+* `server`: Server is the hostname or IP address of the NFS server.
+* `path`: Path that is exported by the NFS server.
+* `readOnly`: A flag indicating whether the storage will be mounted as read only (default false).
+
+NFS doesn't provide an internal provisioner. You need to use an external provisioner to create a StorageClass for NFS.
```
Comment (sftim) [inline content/en/docs/concepts/storage/storage-classes.md:444]:
```suggestion
Kubernetes doesn't include an internal NFS provisioner. You need to use an external provisioner to create a StorageClass for NFS.
```

**Review by sftim — COMMENTED** [review]
I haven't checked this for technical accuracy. I'd be happy to see this merge with or without the change I'm suggesting here.

**Comment by chenxuc** [comment]
@sftim Thanks for the suggestion. I've updated the document with your comment.

**Comment by sftim** [comment]
Thanks for tidying this up @chenxuc

**Pull request merged**